src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -141,17 +141,18 @@ public void launch() {
                 .kv("rootPath", nucleusPaths.rootPath())
                 .kv("configPath", nucleusPaths.configPath()).log("Launch Nucleus");
 
+        final List<DeviceIdentityInterface> provisioningPlugins = findProvisioningPlugins();
+        // Must be called before everything else so that these are available to be
+        // referenced by main/dependencies of main
+        final Queue<String> autostart = findBuiltInServicesAndPlugins(); //NOPMD
+        loadPlugins();
```

Comment:
i think a comment would be useful explaining why this needs to happen before startup

Reply:
This is opened just for testing something and is temporary branch

Reply:
Ah okay, carry on :)